src/common/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { PropsWithChildren, ReactNode } from 'react';
+import { Trans } from 'react-i18next';
+import classNames from 'classnames';
+
+import styles from './Checkbox.module.css';
+
+type Props = PropsWithChildren<{
+  onChange: () => void;
+  label?: string;
+  useTransComponent?: boolean;
+  components?: Array<ReactNode>;
+  i18Key?: string;
+  className?: string;
+}>;
+
+function Checkbox(props: Props) {
+  return (
+    <div className={classNames(styles.container, props.className)}>
+      <input
+        className={styles.checkbox}
+        type="checkbox"
+        onChange={props.onChange}
+      />
+      <span className={styles.label}>
+        {!props.useTransComponent && props.label}
+        {props.useTransComponent && (
```

Comment:
Could `label` include `useTransComponent`, `components` and `i18Key`? So that the type would be `label: string | React.ReactNode` and usage `<Checkbox label="foo" />` or `<Checkbox label={<Trans i18Key... />}`

Reply:
This approach is much better. Changed.